packages/contracts-bedrock/test/LivenessGuard.t.sol

```diff
@@ -0,0 +1,318 @@
+
+    function ownersBeforeLength() public view returns (uint256) {
+        return ownersBefore.length();
+    }
+}
+
+contract LivenessGuard_TestInit is Test, SafeTestTools {
+    using SafeTestLib for SafeInstance;
+
+    event OwnerRecorded(address owner);
+
+    uint256 initTime = 10;
+    WrappedGuard livenessGuard;
+    SafeInstance safeInstance;
+
+    /// @dev Sets up the test environment
+    function setUp() public {
+        vm.warp(initTime);
+        safeInstance = _setupSafe();
+        livenessGuard = new WrappedGuard(safeInstance.safe);
+        safeInstance.setGuard(address(livenessGuard));
+    }
+}
+
+contract LivenessGuard_Constructor_Test is LivenessGuard_TestInit {
+    /// @dev Tests that the constructor correctly sets the current time as the lastLive time for each owner
+    function test_constructor_works() external {
+        address[] memory owners = safeInstance.owners;
+        livenessGuard = new WrappedGuard(safeInstance.safe);
+        for (uint256 i = 0; i < owners.length; i++) {
+            assertEq(livenessGuard.lastLive(owners[i]), initTime);
+        }
+    }
+}
+
+contract LivenessGuard_Getters_Test is LivenessGuard_TestInit {
+    /// @dev Tests that the getters return the correct values
+    function test_getters_works() external {
+        assertEq(address(livenessGuard.safe()), address(safeInstance.safe));
+        assertEq(livenessGuard.lastLive(address(0)), 0);
+    }
+}
+
+contract LivenessGuard_CheckTx_TestFails is LivenessGuard_TestInit {
+    /// @dev Tests that the checkTransaction function reverts if the caller is not the Safe
+    function test_checkTransaction_callerIsNotSafe_revert() external {
+        vm.expectRevert("LivenessGuard: only Safe can call this function");
+        livenessGuard.checkTransaction({
+            to: address(0),
+            value: 0,
+            data: hex"00",
+            operation: Enum.Operation.Call,
+            safeTxGas: 0,
+            baseGas: 0,
+            gasPrice: 0,
+            gasToken: address(0),
+            refundReceiver: payable(address(0)),
+            signatures: hex"00",
+            msgSender: address(0)
+        });
+    }
+}
+
+contract LivenessGuard_CheckTx_Test is LivenessGuard_TestInit {
+    using SafeTestLib for SafeInstance;
+
+    /// @dev Tests that the checkTransaction function succeeds
+    function test_checkTransaction_succeeds() external {
+        // Create an array of the addresses who will sign the transaction. SafeTestTools
+        // will generate these signatures up to the threshold by iterating over the owners array.
+        address[] memory signers = new address[](safeInstance.threshold);
+        signers[0] = safeInstance.owners[0];
+        signers[1] = safeInstance.owners[1];
+
+        for (uint256 i; i < signers.length; i++) {
+            // Don't check topic1 so that we can avoid the ugly txHash calculation.
+            vm.expectEmit(false, true, true, true, address(livenessGuard));
```

Comment:
If no tx hash, then we want to make this `expectEmit(address(livenessGuard))` right?

Reply:
Yes. Good catch.

60aeaf31827e8146b1e8c06f1150fa917dbfb950